src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariants.java

```diff
@@ -1075,34 +1077,22 @@ private VariantContext subsetRecord(final VariantContext vc, final boolean prese
             return vc;
         }
 
-        final VariantContextBuilder builder = new VariantContextBuilder(sub);
-
-        // if there are fewer alternate alleles now in the selected VC, we need to fix the PL and AD values
-        GenotypesContext newGC = GATKVariantContextUtils.updatePLsAndAD(sub, vc);
-
-        // since the VC has been subset (either by sample or allele), we need to strip out the MLE tags
-        builder.rmAttribute(GATKVCFConstants.MLE_ALLELE_COUNT_KEY);
-        builder.rmAttribute(GATKVCFConstants.MLE_ALLELE_FREQUENCY_KEY);
+        // fix the PL and AD values if sub has fewer alleles than original vc and remove a fraction of the genotypes if needed
+        final GenotypesContext oldGs = sub.getGenotypes();
+        GenotypesContext newGC = sub.getNAlleles() == vc.getNAlleles() ? oldGs :
```

Comment:
Make newGC final and the put this initialization in a ```else``` statement following the ```if``` below.